internal/backends/mcu/renderer.rs

```diff
@@ -72,11 +75,25 @@ pub fn render_window_frame(
     let mut screen_fill_profiler = profiler::Timer::new_stopped();
 
     let mut line_buffer = vec![background; size.width as usize];
-    while scene.current_line < size.height_length() {
+
+    let dirty_region = scene
+        .dirty_region
+        .intersection(&PhysicalRect { origin: euclid::point2(0, 0), size })
+        .unwrap_or_default();
+
+    while scene.current_line
+        < size
+            .height_length()
+            .min(dirty_region.origin.y_length() + dirty_region.size.height_length())
```

Comment:
Isn't this already covered by the intersection, i.e. doesn't this always evaluate to `dirty_region.origin.y_length() + dirty_region.size.height_length()` ?

Reply:
yes, that's right. good catch.  (i did not have the intersection yet when i wrote this line)